src/PerfCounters_x86.h

```diff
@@ -265,11 +262,45 @@ static void check_for_xen_pmi_bug() {
   }
 }
 
+static void check_for_zen_speclockmap() {
+  // When the SpecLockMap optimization is not disabled, rr will not work
+  // reliably (e.g. it would work fine on a single process with a single
+  // thread, but not more). When the optimization is disabled, the
+  // perf counter for retired lock instructions of type SpecLockMapCommit
+  // (on PMC 0x25) stays at 0.
+  // See more details at https://github.com/mozilla/rr/issues/2034.
+  struct perf_event_attr attr;
+  // 0x25 == RETIRED_LOCK_INSTRUCTIONS - Counts the number of retired locked instructions
+  // + 0x08 == SPECLOCKMAPCOMMIT
+  init_perf_event_attr(&attr, PERF_TYPE_RAW, 0x510825);
+
+  ScopedFd fd = start_counter(0, -1, &attr);
+  if (fd.is_open()) {
+    int atomic = 0;
+    ioctl(fd, PERF_EVENT_IOC_ENABLE, 0);
```

Comment:
Sorry --- to be clear, I think we should keep the `PERF_EVENT_IOC_DISABLE`.

Reply:
It doesn't matter, though, because we read the counter before the lock add, and after, and compare the values. In fact, the check works without PERF_EVENT_IOC_ENABLE.

Reply:
True. In that case, remove the PERF_EVENT_IOC_ENABLE.